Documentation/Storage-Configuration/Object-Storage-RGW/cosi.md

```diff
@@ -0,0 +1,108 @@
+
+```bash
+kubectl apply -k github.com/kubernetes-sigs/container-object-storage-interface-api ## installs CRDs
+kubectl apply -k github.com/kubernetes-sigs/container-object-storage-interface-controller ## installs COSI controller
+```
+
+## Create a Bucket
+To create a bucket, define a BucketClass and a BucketClaim request. These resources are defined as below:
+
+```yaml
+kind: BucketClass
+apiVersion: objectstorage.k8s.io/v1alpha1
+metadata:
+  name: sample-bcc
+driverName: ceph.objectstorage.k8s.io
+deletionPolicy: Delete
+parameters:
+  ObjectStoreUserSecretName: rook-ceph-object-user-my-store-cosi
+  ObjectStoreUserSecretNamespace: rook-ceph
+---
+kind: BucketClaim
+apiVersion: objectstorage.k8s.io/v1alpha1
+metadata:
+  name: sample-bc
+spec:
+  bucketClassName: sample-bcc
+  protocol
+```
+
+The Ceph object store user named `cosi` will be created. The user's credentials are automatically stored in a K8s secret. The secret name is then referenced from the BucketClass.
+
+## Bucket Access
+
+Define access to the bucket by creating a BucketAccessClass and BucketAccess resource:
+
+```yaml
+kind: BucketAccessClass
+apiVersion: objectstorage.k8s.io/v1alpha1
+metadata:
+  name: sample-bac
+driverName: ceph.objectstorage.k8s.io
+authenticationType: KEY
+---
+kind: BucketAccess
+apiVersion: objectstorage.k8s.io/v1alpha1
+metadata:
+  name: sample-access
+spec:
+  bucketAccessClassName: sample-bac
+  bucketClaimName: sample-bc
+  protocol: s3
+  # Change to the name of the secret where access details are stored
+  credentialsSecretName: sample-secret-name
+```
+
+
+The secret will be created which contains the access details for the bucket in JSON format:
+
+```json
+{
+  "metadata": {
+    "name": "bc-81733d1a-ac7a-4759-96f3-fbcc07c0cee9",
+    "creationTimestamp": null
+  },
+  "spec": {
+    "bucketName": "sample-bcc1fc94b04-6011-45e0-a3d8-b6a093055783",
+    "authenticationType": "KEY",
+    "secretS3": {
+      "endpoint": "http://rook-ceph-rgw-my-store.rook-ceph.svc:80",
+      "region": "",
+      "accessKeyID": "LI2LES8QMR9GB5SZLB02",
+      "accessSecretKey": "s0WAmcn8N1eIBgNV0mjCwZWQmJiCF4B0SAzbhYCL"
+    },
+    "secretAzure": null,
+    "protocols": [
+      "s3"
+    ]
+  }
+}
+```
+
+## Consuming the Bucket via secret
+
+To access the bucket from an application pod, mount the secret for accessing the bucket:
+
+```yaml
+  volumes:
+  - name: cosi-secrets
+    secret:
+      #  Set the name of the secret from the BucketAccess
+      secretName: sample-secret-name
+  spec:
+    containers:
+    - name: sample-app
+      volumeMounts:
+      - name: cosi-secrets
+        mountPath: /data/cosi
+```
+
+The Secret will be mounted in the pod in the path: `/data/cosi/BucketInfo`. The app must parse the JSON object to load the bucket connection details.
```

Comment:
Is there not a way to mount the info as env vars? If not, we can consider taking this this as a feature/doc request for COSI upstream. Many legacy apps will proably use env vars.

Reply:
Not directly atm, but we can do it via init container, secret contains as entire file

Reply:
It might be good to add an example of the init container method in a future update